### PR TITLE
Wire wallet seed with seed screen

### LINKED ISSUE #101: Wire wallet seed with seed screen
Currently the seed is mocked, it should be wired with the seed phrase used to generate the BIP84 wallet.

Acceptance Criteria

- Seed Phrase is fetched from the rust lib

At some point we should think about storing the seed file to the mobile device, but this is not focus of this issue.

### PR DESCRIPTION
**Pull request by holzeis**
resolves #101

### CODE REVIEW

File: integration_test/dashboard_test.dart
Comment (da-kami) [inline integration_test/dashboard_test.dart:91]: Does this mean that this test has a dependency on `testnet`, i.e. it actually requires an internet connection and fetches from a public node? 

I think it would be great if these tests would not depend on external things. Not super important for now.
On a more general note: For this test the actual seed really should not matter, why do we even need the api call here?

File: lib/seed.dart
Comment (da-kami) [inline lib/seed.dart:30]: ❓ Can it happen that this call never finishes or can we be sure that the seed is available through the API?

**Review by da-kami — APPROVED** [review]

File: integration_test/dashboard_test.dart
Comment (klochowicz) [reply integration_test/dashboard_test.dart:91]: maybe we could run 'signet' on this? (and not connect with any external infrastructure?). Alternatively, we could allow some custom `None` network where our wallet would be hardcoded to some value and not using any public infrastructure.

File: lib/main.dart
Comment (klochowicz) [inline lib/main.dart:1]: thank you for the great commit message! I've learned something new today 👍

File: lib/seed.dart
Comment (klochowicz) [inline lib/seed.dart:26]: I guess it's good for now, we can always refactor it later

File: rust/src/api.rs
Comment (klochowicz) [inline rust/src/api.rs:27]: yeah, it's sent over once, so it's not a problem at all.

sometimes boxing (ie. wrapping in a box, like : `Result<Box<Type>>`) can help, as it just passes a pointer to  data already on the heap, but I don't think it's worth it here

File: rust/src/wallet.rs
Comment (klochowicz) [inline rust/src/wallet.rs:30]: are you planning to address my earlier comments from the initial seed review? I would prefer if we didn't need to keep strings around, so we don't accidentally print them out in tracing logs etc.... (can be done as a follow-up, no need to block this PR)

File: rust/src/wallet.rs
Comment (klochowicz) [inline rust/src/wallet.rs:102]: 🛠 might be a good time to extract this boilerplate into a free function

File: integration_test/dashboard_test.dart
Comment (holzeis) [reply integration_test/dashboard_test.dart:91]: > Does this mean that this test has a dependency on `testnet`, i.e. it actually requires an internet connection and fetches from a public node?

Yes it does, but its not different for any other test that is running the TenTenOneApp, as the wallet will be initialised there.  It's only done here explicitly, because the dashboard_test are running a different app config to be able to access the providers. 

> On a more general note: For this test the actual seed really should not matter, why do we even need the api call here?

Well, because the tests are failing otherwise due to a rust error. The seed is trying to access an uninitialised wallet. Once the logging PR is merged I will adapt and add an error handling.

File: lib/seed.dart
Comment (holzeis) [reply lib/seed.dart:30]: > Can it happen that this call never finishes
It could, but it wouldn't block the screen.

> can we be sure that the seed is available through the API?
No we can't, e.g. if the wallet is not initialised this call will fail. (that's the reason for the init wallet in the integration tests).

Error handling should be there, but flutter complains about simply printing the error so I opened the logging PR to be able to log errors properly. I will adapt here once https://github.com/itchysats/10101/pull/104 is merged.

File: rust/src/wallet.rs
Comment (holzeis) [reply rust/src/wallet.rs]: Yes 🙂 https://github.com/itchysats/10101/pull/102/commits/b21679b29dcc501066a810b0267aded26b755ab1

File: integration_test/dashboard_test.dart
Comment (holzeis) [reply integration_test/dashboard_test.dart]: @da-kami https://github.com/itchysats/10101/pull/102/commits/451f23039bf77b019f914b33f5a358c2904dfda0

File: lib/seed.dart
Comment (holzeis) [reply lib/seed.dart:32]: Added error handling https://github.com/itchysats/10101/pull/102/commits/451f23039bf77b019f914b33f5a358c2904dfda0

File: rust/src/wallet.rs
Comment (holzeis) [reply rust/src/wallet.rs]: https://github.com/itchysats/10101/pull/102/commits/510d5d4eed1bdcfd082f8217f3b0e0c1e918c0c5

File: rust/src/seed.rs
Comment (klochowicz) [inline rust/src/seed.rs:44]: nit: you can probably use `word_iter()` function

**Review by klochowicz — APPROVED** [review]
nice one! thanks for addressing the comments :-)

**Comment by klochowicz** [comment]
@holzeis I hope you don't mind that I autosquashed the fixup commits to allow borsing :)

**Pull request merged**
